cmake/Findkeyutils.cmake

```diff
@@ -0,0 +1,27 @@
+# Try to find Keyutils
+# Once done, this will define
+#
+# KEYUTILS_FOUND - system has keyutils
+# KEYUTILS_INCLUDE_DIR - the keyutils include directories
+# KEYUTILS_LIBRARIES - link these to use keyutils
+
+find_path(KEYUTILS_INCLUDE_DIR keyutils.h PATHS
```

Comment:
keyutils (on my system at least) has a pkgconfig file.
We should wrap that rather than manual calls where possible, it's more future proof

Reply:
`pkg_check_modules(Keyutils IMPORTED_TARGET libkeyutils)` would suffice.